File: web/webTrung/java_spring_-thymeleaf_food/web/src/main/java/doan/_java_food/models/Order.java
package doan._java_food.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Table(name = "orders")
@Getter
@Setter
public class Order {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name =  "code")
    private String code;

    @Column(name =  "status")
    private String status;

    @Column(name =  "payment_status")
    private String paymentStatus;

    @Column(name =  "payment_type")
    private Integer paymentType;


    @Column(name = "type", columnDefinition = "VARCHAR(255) DEFAULT 'BOOKING'")
    private String type;

    @Column(columnDefinition = "TEXT",name =  "receiver_address")
    private String receiverAddress;

    @Column(name =  "receiver_email")
    private String receiverEmail;

    @Column(name =  "receiver_name")
    private String receiverName;

    @Column(name =  "receiver_phone")
    private String receiverPhone;

    @Column(columnDefinition = "TEXT", name =  "note")
    private String note;

    @Column(name = "total_price")
    private String totalPrice;

    @Column(name = "discount")
    private Float discount;

    @Column(name = "user_id")
    private Integer userId;

    @Column(name = "restaurant_name")
    private String restaurantName;

    @Column(name = "restaurant_id")
    private Integer restaurantId;

    @Column( name =  "logo")
    private String logo;

    @Column(name = "price")
    private String price;

    @Column(name = "time_booking")
    private LocalDateTime timeBooking;

    @Column(name = "quantity")
    private Integer quantity;

//    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
//    @JoinColumn(name = "order_id")
//    private List<TransactionEntity> transactions;

    @Column(name = "created_at")
    private Date createdAt;

    @Column(name = "updated_at")
    private Date updatedAt;

    public Float getDiscount() {
        return discount;
    }

    public void setDiscount(Float discount) {
        this.discount = discount;
    }
}

File: app/pasgo_restaurants/lib/domain/entities/order/order.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pasgo_restaurants_ecommerce/data/models/restaurant/uu_model.dart';

part 'order.g.dart';

@JsonSerializable()
class OrderM {
  final int id;
  final String code;
  final String status;
  final String type;
  // final String receiverAddress;
  final String? receiverName;
  final String receiverEmail;
  final String? receiverPhone;
  final String? note;
  // final double totalPrice;
  final double? discount;
  final String? restaurantName;
  final String? logo;
  // final String? price;
  final String timeBooking;
  final double quantity;
  final String createdAt;
  final String updatedAt;
  final UUserModel user;

  const OrderM({
    required this.id,
    required this.code,
    required this.status,
    required this.type,
    // required this.receiverAddress,
    required this.receiverEmail,
    this.receiverName,
    this.receiverPhone,
    this.note,
    // required this.totalPrice,
    this.discount,
     this.restaurantName,
    this.logo,
    // required this.price,
    required this.timeBooking,
    required this.quantity,
    required this.createdAt,
    required this.updatedAt,
    required this.user,
  });

  factory OrderM.fromJson(Map<String, dynamic> json) => _$OrderMFromJson(json);

  Map<String, dynamic> toJson() => _$OrderMToJson(this);
}
